apps/etterlatte-saksbehandling-ui/client/src/components/behandling/beregningsgrunnlag/BeregningsgrunnlagBarnepensjon.tsx

```diff
@@ -122,171 +69,138 @@ const BeregningsgrunnlagBarnepensjon = (props: { behandling: IBehandlingReducer
+    }
+    lagreBeregningsgrunnlagRequest(
+      {
+        behandlingId: behandling.id,
+        grunnlag,
+      },
+      () => dispatch(oppdaterBeregingsGrunnlag(grunnlag))
     )
   }
 
-  const onSubmit = () => {
-    if (skalViseSoeskenjustering && !(soeskenGrunnlagsData || behandling.beregningsGrunnlag?.soeskenMedIBeregning)) {
-      setSoeskenJusteringMangler(true)
+  const oppdaterSoeskenJustering = (soeskenGrunnlag: Soeskengrunnlag) => {
+    const grunnlag = {
+      ...behandling.beregningsGrunnlag,
+      soeskenMedIBeregning: mapListeTilDto(soeskenGrunnlag),
+      institusjonsopphold: behandling.beregningsGrunnlag?.institusjonsopphold,
+      begegningsmetodeFlereAvdoede: behandling.beregningsGrunnlag?.begegningsmetodeFlereAvdoede,
+      beregningsMetode: behandling.beregningsGrunnlag?.beregningsMetode ?? {
+        beregningsMetode: BeregningsMetode.NASJONAL,
+      },
     }
-    if (behandling.beregningsGrunnlag?.soeskenMedIBeregning || soeskenGrunnlagsData || !skalViseSoeskenjustering) {
-      dispatch(resetBeregning())
-      const beregningsgrunnlag = {
-        soeskenMedIBeregning: soeskenGrunnlagsData
-          ? mapListeTilDto(soeskenGrunnlagsData)
-          : behandling.beregningsGrunnlag?.soeskenMedIBeregning ?? [],
-        institusjonsopphold: institusjonsoppholdsGrunnlagData
-          ? mapListeTilDto(institusjonsoppholdsGrunnlagData)
-          : behandling.beregningsGrunnlag?.institusjonsopphold ?? [],
-        beregningsMetode: beregningsMetodeBeregningsgrunnlag
-          ? beregningsMetodeBeregningsgrunnlag
-          : behandling.beregningsGrunnlag?.beregningsMetode ?? {
-              beregningsMetode: BeregningsMetode.NASJONAL,
-            },
-        begegningsmetodeFlereAvdoede: beregningsmetodeForAvdoede
-          ? mapListeTilDto(beregningsmetodeForAvdoede)
-          : behandling.beregningsGrunnlag?.begegningsmetodeFlereAvdoede ?? [],
-      }
 
-      postBeregningsgrunnlag(
-        {
-          behandlingId: behandling.id,
-          grunnlag: beregningsgrunnlag,
-        },
-        () =>
-          postOpprettEllerEndreBeregning(behandling.id, (beregning: Beregning) => {
-            dispatch(oppdaterBeregingsGrunnlag(beregningsgrunnlag))
-            dispatch(oppdaterBehandlingsstatus(IBehandlingStatus.BEREGNET))
-            dispatch(oppdaterBeregning(beregning))
-            next()
-          })
-      )
-    }
+    lagreBeregningsgrunnlagRequest(
+      {
+        behandlingId: behandling.id,
+        grunnlag,
+      },
+      () => dispatch(oppdaterBeregingsGrunnlag(grunnlag))
+    )
   }
 
+  useEffect(() => {
+    hentBeregningsgrunnlagRequest(behandling.id, (result) => {
+      if (result) {
+        dispatch(
+          oppdaterBeregingsGrunnlag({ ...result, institusjonsopphold: result.institusjonsoppholdBeregningsgrunnlag })
+        )
+      }
+      hentTrygdetiderRequest(behandling.id)
+    })
+  }, [])
```

Comment:
denne bør flyttes lengre opp, til rett etter useState for manglendeSoeskenjustering